KeyVault.Acmebot/Startup.cs

```diff
@@ -93,6 +104,11 @@ public override void Configure(IFunctionsHostBuilder builder)
                     return new GratisDnsProvider(options.GratisDns);
                 }
 
+                if (options.TransIp != null)
+                {
+                    return new TransIpProvider(options, options.TransIp, credential);
```

Comment:
I'd like to see the handling of `TokenCredential` changed so that each of them is created in their own constructor, just like `AzureDnsProvider`.